code/modules/mob/dead/observer/observer.dm

```diff
@@ -136,15 +136,15 @@
 	return ghost
 
 
-/mob/proc/set_away_time()
+/mob/proc/set_away_time(new_away)
 	return
 
-/mob/living/set_away_time()
+/mob/living/set_away_time(new_away = world.time)
 	away_time = world.time //Generic way to handle away time, currently unused.
 
 
-/mob/living/carbon/Xenomorph/set_away_time()
-	away_time = -XENO_AFK_TIMER //Xenos who force-ghost can be immediately taken by observers.
+/mob/living/carbon/Xenomorph/set_away_time(new_away = -XENO_AFK_TIMER)
+	away_time = new_away //Xenos who force-ghost can be immediately taken by observers.
```

Comment:
```suggestion
	away_time = new_away //Xenos who force-ghost can be immediately taken by observers.
	handle_afk_takeover()
```
Will require to move `set_away_time()` in `/mob/proc/ghostize(` to the end of the proc, after the key transition happens to guarantee the client is gone by then, but would allow for instant takeover of ghostized xenos as well.

Reply:
Does this really change anything? On next tick the xeno would do this check regardless?

Reply:
It won't, it will only be called after `XENO_AFK_TIMER` instead of immediately.

Reply:
Reminder that this is still a thing.